tests/framework/utils.py

```diff
@@ -715,14 +717,17 @@ def start_screen_process(screen_log, session_name, binary_path, binary_params):
     # Configure screen to flush stdout to file.
     run_cmd(FLUSH_CMD.format(session=session_name))
 
-    children_count = len(screen_ps.children())
-    if children_count != 1:
-        raise RuntimeError(
-            f"Failed to retrieve child process id for binary '{binary_path}' "
-            f"screen session process had [{children_count}]"
-        )
+    if not ignore_binary_pid:
+        children_count = len(screen_ps.children())
+        if children_count != 1:
+            raise RuntimeError(
+                f"Failed to retrieve child process id for binary '{binary_path}' "
+                f"screen session process had [{children_count}]"
+            )
+
+        return screen_pid, screen_ps.children()[0].pid
```

Comment:
If there will be no users of this, it would be better to just remove this

Reply:
Yeah if there are no users it's better to remove. The caller can implement that if needed.